components/match2/wikis/pokemon/match_summary.lua

```diff
@@ -0,0 +1,316 @@
+			banData[1].color = extradata.team1side
+			banData[2].color = extradata.team2side
+			banData.numberOfBans = numberOfBans
+			championBanData[gameIndex] = banData
+		end
+	end
+
+	-- Add the Champion Bans
+	if not Table.isEmpty(championBanData) then
+		local championBan = ChampionBan()
+
+		for gameIndex, banData in ipairs(championBanData) do
+			championBan:banRow(banData, gameIndex, banData.numberOfBans, match.date)
+		end
+
+		body:addRow(championBan)
+	end
+
+	return body
+end
+
+function CustomMatchSummary._createGame(game, gameIndex, date)
+	local row = MatchSummary.Row()
+	local extradata = game.extradata or {}
+
+	local championsData = {{}, {}}
+	local championsDataIsEmpty = true
+	for champIndex = 1, _NUM_CHAMPIONS_PICK do
+		if String.isNotEmpty(extradata['team1champion' .. champIndex]) then
+			championsData[1][champIndex] = extradata['team1champion' .. champIndex]
+			championsDataIsEmpty = false
+		end
+		if String.isNotEmpty(extradata['team2champion' .. champIndex]) then
+			championsData[2][champIndex] = extradata['team2champion' .. champIndex]
+			championsDataIsEmpty = false
+		end
+		championsData[1].color = extradata.team1side
+		championsData[2].color = extradata.team2side
+	end
+
+	if
+		Table.isEmpty(game.scores) and
+		String.isEmpty(game.winner) and
+		championsDataIsEmpty
+	then
+		return nil
+	end
+
+	row	:addClass('brkts-popup-body-game')
+		:css('font-size', '85%')
+		:css('overflow', 'hidden')
+
+	local score
+	if not Table.isEmpty(game.scores) then
+		score = table.concat(game.scores, '-')
+	end
+
+	row:addElement(CustomMatchSummary._opponentChampionsDisplay(championsData[1], _NUM_CHAMPIONS_PICK, date, false))
+	row:addElement(CustomMatchSummary._createCheckMark(game.winner == 1))
+	row:addElement(mw.html.create('div')
+		:addClass('brkts-popup-body-element-vertical-centered')
+		:wikitext(CustomMatchSummary._createAbbreviation{
+			title = String.isEmpty(game.length) and ('Game ' .. gameIndex .. ' picks') or 'Map Scores',
+			text = score or ('Game ' .. gameIndex),
+		})
+	)
+	row:addElement(CustomMatchSummary._createCheckMark(game.winner == 2))
+	row:addElement(CustomMatchSummary._opponentChampionsDisplay(championsData[2], _NUM_CHAMPIONS_PICK, date, true))
+
+	-- Add Comment
+	if not Logic.isEmpty(game.comment) then
+		row:addElement(MatchSummary.Break():create())
+		local comment = mw.html.create('div')
+		comment :wikitext(game.comment)
+				:css('margin', 'auto')
+		row:addElement(comment)
+	end
+
+	return row
+end
+
+function CustomMatchSummary._createCheckMark(isWinner)
+	local container = mw.html.create('div')
+		:addClass('brkts-popup-spaced')
+		:css('line-height', '27px')
+		:css('margin-left', '3%')
+		:css('margin-right', '3%')
+
+	if isWinner then
+		container:node(_GREEN_CHECK)
+	else
+		container:node(_NO_CHECK)
+	end
+
+	return container
+end
+
+function CustomMatchSummary._createAbbreviation(args)
+	return '<i><abbr title="' .. args.title .. '">' .. args.text .. '</abbr></i>'
+end
```

Comment:
Can Module:Abbreviation be used instead?

Reply:
the `<i></i>` around it is not done by the abbr module, but rest is^^

Reply:
<i> doesn't belong in the abbr code either way, but outside it :D

Reply:
What should I edit here or keep it?

Reply:
Remove this function & use Module:Abbreviation to create your Abbreviations instead

Reply:
> Remove this function & use Module:Abbreviation to create your Abbreviations instead

any code examples on it? since im not sure that just remove those three line and add **local Abbreviation = require('Module:Abbreviation')** would be just that

Reply:
yeah use that at the top to import the module

and when you want an Abbreviation you just use
`Abbreviation.make(text, title)`

Reply:
> yeah use that at the top to import the module
> 
> and when you want an Abbreviation you just use `Abbreviation.make(text, title)`

Ok, I have updated the new revision in this Pull, but its incomplete, what currently happens is the either "Game X" or the total score in the middle is now absent.
 
![image](https://user-images.githubusercontent.com/88981446/183125305-ff12b08f-fd65-459d-8d89-fd12ad8c1ede.png)

Reply:
pls link the example page and i can look into it tomorrow

Reply:
> pls link the example page and i can look into it tomorrow

https://liquipedia.net/pokemon/User:Hesketh2/Test

Reply:
https://liquipedia.net/pokemon/index.php?title=Module%3AMatchSummary&type=revision&diff=138565&oldid=138455